### PR TITLE
EVAKA-4240 citizen reservation inline editor

### PR DESCRIPTION
**Pull request by jaahon**
#### Summary

Add inline editor to citizen reservation day view where single day reservations can be updated quickly.
Also,
* Show month with first available end date when opening date picker in reservation modal
* Fix a bug in citizen reservation overwrite
* Show same reservations only once

### CODE REVIEW

File: frontend/src/citizen-frontend/calendar/DayView.tsx [hunk 1]
```diff
@@ -40,53 +60,256 @@ export default React.memo(function DayView({ date, data, selectDate }: Props) {
...
+                  )}`}</span>
+                )}
+                <Label>Toteuma</Label>
+                <span>–</span>
+              </Grid>
+            </Fragment>
+          ))}
+      {confirmationModal ? (
+        <InfoModal
+          title="Haluatko tallentaa muutokset?"
```
Comment (mikkopiu) [inline frontend/src/citizen-frontend/calendar/DayView.tsx:170]: Shouldn't these be localized? These will after all be read by screenreaders.

**Review by mikkopiu — CHANGES_REQUESTED** [review]
Missing some localizations and IMO the logic here is complex enough that some E2E tests would be nice to have.

File: frontend/src/citizen-frontend/calendar/DayView.tsx (hunk 1 above)
Comment (jaahon) [reply frontend/src/citizen-frontend/calendar/DayView.tsx:170]: Yeah, everything should be localized when it's enabled in production. And this is an actual "pop up" modal so it's definitely visible to everyone.

**Comment by jaahon** [comment]
> Missing some localizations and IMO the logic here is complex enough that some E2E tests would be nice to have.

We are planning to write e2e-tests when everything's finished and ready for production.

**Review by mikkopiu — APPROVED** [review]
OK with above comment replies

**Review by Joosakur — APPROVED** [review]

**Pull request merged**